webproject/src/main/java/com/google/sps/dataManagers/MessagesManager.java

```diff
@@ -0,0 +1,65 @@
+package com.google.sps.dataManagers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.ServletException;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseReference;
+import com.google.sps.data.MessageProto.Message;
+import com.google.sps.util.FirebaseUtil;
+import com.google.sps.util.TimestampUtil;
+
+@Singleton
+public class MessagesManager {
+    private FirebaseUtil firebaseUtil;
+    private UserRoomManager userRoomManager;
+
+    @Inject
+    public MessagesManager(FirebaseUtil firebaseUtil, UserRoomManager userRoomManager) {
+        this.firebaseUtil = firebaseUtil;
+        this.userRoomManager = userRoomManager;
+    }
+
+    public List<Message> getMessages(String roomId) throws ServletException {
+        if (!userRoomManager.isRoomIdValid(roomId)) {
+            throw new ServletException("Invalid roomId");
+        }
+
+        DatabaseReference ref = firebaseUtil.getMessagesReference().child(roomId);
+        List<DataSnapshot> dataSnapshots = firebaseUtil.getAllSnapshotsFromReference(ref);
+
+        List<Message> messages = new ArrayList<>();
+        for (DataSnapshot dataSnapshot : dataSnapshots) {
+            String user = dataSnapshot.child("user").getValue(String.class);
```

Comment:
move to message class

Reply:
changed it to use stream to get the list of messages

Reply:
changed it to use stream to get a list of messages